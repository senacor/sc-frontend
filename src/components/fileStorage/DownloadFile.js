import React from 'react';
import Icon from '@material-ui/core/Icon';
import IconButton from '@material-ui/core/IconButton';
import * as actions from '../../actions';
import { getDownloadedFile } from '../../reducers/selector';
import { connect } from 'react-redux';

export class DownloadFile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      employeeId: props.employeeId,
      fileId: props.fileId
    };
  }

  handleClick = (employeeId, fileId) => () => {
    this.props.downloadFile(employeeId, fileId);
  };

  checkDownloadFile(file) {
    let { employeeId, fileId } = this.state;
    return file.id && file.employeeId === employeeId && file.fileId === fileId;
  }
  render() {
    let { employeeId, fileId } = this.state;

    if (this.checkDownloadFile(this.props.downloadedFile)) {
      this.props.resetDownloadedFile();
      window.open(this.props.downloadedFile.url);
    }

    return (
      <IconButton onClick={this.handleClick(employeeId, fileId)}>
        <Icon>get_app</Icon>
      </IconButton>
    );
  }
}

export default connect(
  state => ({
    downloadedFile: getDownloadedFile(state)
  }),
  {
    downloadFile: actions.downloadFile,
    resetDownloadedFile: actions.resetDownloadedFile
  }
)(DownloadFile);
