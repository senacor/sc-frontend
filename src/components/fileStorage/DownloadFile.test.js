import { createShallow } from '@material-ui/core/test-utils';
import React from 'react';
import { DownloadFile } from './DownloadFile';

describe('DownloadFile Component', () => {
  let shallow = createShallow({ dive: true });

  it('should match snapshot', () => {
    let component = shallow(
      <DownloadFile employeeId={504} fileId={42} downloadedFile={{}} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should not start an download', () => {
    let downloadedFile = {
      id: 21,
      employeeId: 504,
      fileId: 999,
      url: 'https://url.com'
    };
    let resetDownloadedFileMock = jest.fn();
    let openMock = jest.fn();

    global.window = jest.fn();
    global.window.open = openMock;

    shallow(
      <DownloadFile
        employeeId={504}
        fileId={42}
        downloadedFile={downloadedFile}
        resetDownloadedFile={resetDownloadedFileMock}
      />
    );
    expect(resetDownloadedFileMock).toHaveBeenCalledTimes(0);
    expect(openMock).toHaveBeenCalledTimes(0);
  });
});
