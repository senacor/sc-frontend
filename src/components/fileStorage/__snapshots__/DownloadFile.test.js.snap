// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadFile Component should match snapshot 1`] = `
<IconButton
  classes={
    Object {
      "colorInherit": "MuiIconButton-colorInherit-2",
      "colorPrimary": "MuiIconButton-colorPrimary-3",
      "colorSecondary": "MuiIconButton-colorSecondary-4",
      "disabled": "MuiIconButton-disabled-5",
      "label": "MuiIconButton-label-6",
      "root": "MuiIconButton-root-1",
    }
  }
  color="default"
  disabled={false}
  onClick={[Function]}
>
  <WithStyles(Icon)>
    get_app
  </WithStyles(Icon)>
</IconButton>
`;

exports[`DownloadFile Component should match snapshot if file is downloaded 1`] = `
<IconButton
  classes={
    Object {
      "colorInherit": "MuiIconButton-colorInherit-2",
      "colorPrimary": "MuiIconButton-colorPrimary-3",
      "colorSecondary": "MuiIconButton-colorSecondary-4",
      "disabled": "MuiIconButton-disabled-5",
      "label": "MuiIconButton-label-6",
      "root": "MuiIconButton-root-1",
    }
  }
  color="default"
  disabled={false}
  onClick={[Function]}
>
  <WithStyles(Icon)>
    get_app
  </WithStyles(Icon)>
</IconButton>
`;
